### PR TITLE
i#3544 RV64: Implements more functions and enable more tests

### PR DESCRIPTION
**Pull request by ksco**
Added a few more tests to `RUNS_ON_QEMU` and more tests (like `linux.signalXXXX`) is passing under real hardware, but not on QEMU for some reason unknown for now.

Issue: #3544

### CODE REVIEW

**Comment by ksco** [comment]
It's been a while since my last contribution, is the CI getting a lot faster?!

**Comment by derekbruening** [comment]
> It's been a while since my last contribution, is the CI getting a lot faster?!

There were some timeouts due to hangs on the aarch64 tester now fixed; the 32-bit previously-slowest job is temporarily shrunk to get things green (still many AMD 32-bit failures #6417; 32-bit x86 just hasn't been high priority...).

File: core/arch/riscv64/emit_utils.c [hunk 1]
```diff
@@ -540,8 +540,39 @@ cbr_fallthrough_exit_cti(cache_pc prev_cti_pc)
...
+     * See insert_exit_stub_other_flags(), the last instruction in indirect exit stub will
+     * always be a c.nop.
+     */
+    pc = (uint *)(stub_pc +
+                  exit_stub_size(dcontext, ibl_code->indirect_branch_lookup_routine,
+                                 f->flags) -
+                  RISCV64_INSTR_COMPRESSED_SIZE);
+    pc = get_stub_branch(pc) - 1;
+
+    ASSERT(get_ibl_entry_tls_offs(dcontext, exit_target) <= (2 << 11) - 1);
```
Comment (derekbruening) [inline core/arch/riscv64/emit_utils.c:565]: 2 because targets are always aligned to 2?

File: core/arch/riscv64/emit_utils.c [hunk 2]
```diff
@@ -540,8 +540,39 @@ cbr_fallthrough_exit_cti(cache_pc prev_cti_pc)
...
+    /* Format of the ld instruction:
+        | imm[11:0] |  rs1  |011|  rd  |0000011|
+        ^   31-20   ^ 19-15 ^   ^ 11-7 ^
+     */
+    /* ld a1, offs(reg_stolen) */
+    *(uint *)vmcode_get_writable_addr((byte *)pc) = 0x3003 |
+        get_ibl_entry_tls_offs(dcontext, exit_target) << 20 |
+        (dr_reg_stolen - DR_REG_ZERO) << 15 | (DR_REG_A1 - DR_REG_ZERO) << 7;
```
Comment (derekbruening) [inline core/arch/riscv64/emit_utils.c:573]: Share lines 565..573 with link_indirect_exit_arch()?

Comment (derekbruening) [inline core/arch/riscv64/emit_utils.c:573]: Named constants would help, esp when the same ones like 0x3003 are repeated.

**Review by derekbruening — APPROVED** [review]

File: core/arch/riscv64/emit_utils.c (hunk 1 above)
Comment (ksco) [reply core/arch/riscv64/emit_utils.c]: It's a typo, fixed.

File: core/arch/riscv64/emit_utils.c (hunk 2 above)
Comment (ksco) [reply core/arch/riscv64/emit_utils.c]: Added `set_indirect_exit_target()` to share common code.

Comment (ksco) [reply core/arch/riscv64/emit_utils.c]: Make it a constant, same for the sd instruction in `unpatch_stub()`.

**Pull request merged**